src/structs/probability_tables.rs

```diff
@@ -319,26 +332,40 @@ impl ProbabilityTables {
         };
     }
 
-    fn estimate_dir_average(
-        dc_estimates: &[i16; 16],
-        dc_start_index: usize,
-        min_dc: &mut i16,
-        max_dc: &mut i16,
-    ) -> i32 {
+    fn estimate_dir_average(dc_estimates: &[i16; 8], min_dc: &mut i16, max_dc: &mut i16) -> i32 {
         let mut dir_average: i32 = 0;
         for i in 0..8 {
-            let cur_est = dc_estimates[dc_start_index + i];
-            dir_average += i32::from(cur_est);
-
-            if *min_dc > cur_est {
-                *min_dc = cur_est;
-            }
-
-            if *max_dc < cur_est {
-                *max_dc = cur_est;
-            }
+            let cur_est = dc_estimates[i];
+            dir_average += cur_est as i32;
         }
 
+        *min_dc = min(
```

Comment:
branchless?

Reply:
the compiler does something amazing here:

        movdqu  xmm0, xmmword ptr [rdi]
        pshufd  xmm1, xmm0, 238
        pminsw  xmm1, xmm0
        pshufd  xmm0, xmm1, 85
        pminsw  xmm0, xmm1
        movdqa  xmm1, xmm0
        psrld   xmm1, 16
        pminsw  xmm1, xmm0
        movd    eax, xmm1